Source/Core/check.js

```diff
@@ -0,0 +1,111 @@
+/*global define*/
+define([
+    './defined',
+    './DeveloperError',
+    './isArray'
+], function(defined,
+            DeveloperError,
+            isArray) {
+    'use strict';
+
+    var exports = {
```

Comment:
Probably explicitly mark this `@private` as I don't think it is worth increasing the API surface area for.